File: 백준/Silver/1991. 트리 순회/트리 순회.py
import sys
 
N = int(sys.stdin.readline().strip())
graph = {}
 
for n in range(N):
    root, dest1, dest2 = sys.stdin.readline().strip().split()
    graph[root] = [dest1, dest2]
 
 
def preorder(root):
    if root != '.':
        print(root, end='')
        preorder(graph[root][0])  
        preorder(graph[root][1])  
 
 
def inorder(root):
    if root != '.':
        inorder(graph[root][0])
        print(root, end='')  
        inorder(graph[root][1])  
 
 
def postorder(root):
    if root != '.':
        postorder(graph[root][0])  
        postorder(graph[root][1])  
        print(root, end='')  
 
 
preorder('A')
print()
inorder('A')
print()
postorder('A')

File: 백준/Silver/1991. 트리 순회/트리 순회.cc
#include <iostream>
#include <algorithm>
#include <string>
#include <map>
using namespace std;
map<string,pair<string,string> > mymap;
void preorder(string root){
    if(root == "."){
        return;
    }
    cout << root;
    preorder(mymap[root].first);
    preorder(mymap[root].second);
}
void inorder(string root){
    if(root == "."){
        return;
    }
    inorder(mymap[root].first);
    cout << root;
    inorder(mymap[root].second);
}
void postorder(string root){
    if(root == "."){
        return;
    }
    postorder(mymap[root].first);
    postorder(mymap[root].second);
    cout << root;
}
int main() {
    int n;
    cin >> n;
    for(int i = 0;i<n;i++){
        string root, left,right;
        cin >> root >> left >> right;
        mymap[root] = make_pair(left,right);
    }
    preorder("A");
    cout << '\n';
    inorder("A");
    cout << '\n';
    postorder("A");
    return 0;
}